src/model.js

```diff
@@ -3513,16 +3509,27 @@ Instead of specifying a Model, either:
    * @returns {object}
    */
   where(checkVersion) {
+    if (this.constructor.primaryKeyAttributes.length === 0) {
+      throw new Error(
+        `This model instance method needs to be able to identify the entity in a stable way, but the model does not have a primary key attribute definition. Either add a primary key to this model, or use one of the following alternatives:
+  
+      - instance methods "save", "update", "decrement", "increment": Use the static "update" method instead.
+      - instance method "reload": Use the static "findOne" method instead.
+      - instance methods "destroy" and "restore": use the static "destroy" and "restore" methods instead.
+        `.trim(),
+      );
+    }
+
     const where = this.constructor.primaryKeyAttributes.reduce((result, attribute) => {
+      if (_.isNil(attribute)) {
```

Comment:
`attribute` is the name of the attribute, you need to check whether its value is nullish (which is returned by `this.get(attribute, { raw: true });`)